include/app_memory/partitions.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2019 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_APP_MEMORY_PARTITIONS_H
+#define ZEPHYR_APP_MEMORY_PARTITIONS_H
+
+#ifdef CONFIG_USERSPACE
+#include <kernel.h> /* For struct k_mem_partition */
+
+#if defined(CONFIG_MBEDTLS)
+extern struct k_mem_partition k_mbedtls_partition;
```

Comment:
Is mbedtls enough of a core thing that it belongs in a single global header like this?  Seems aesthetically that it really ought to be exported by a mbedtls header.  Meh.

Reply:
mbedtls headers are from upstream project. we don't want to change those, this is very specific to Zephyr.

I'm going to put other partitions in here too as we create them.

Reply:
There are headers that we can change to tuning mbedTLS options in  ./ext/lib/crypto/mbedtls/configs.
Though I've no complain with this new header since it will be used by other partitions.

Reply:
> There are headers that we can change to tuning mbedTLS options in ./ext/lib/crypto/mbedtls/configs

It wouldn't be appropriate to define this there.
For one thing, there are a lot of different config headers and we'd have to change all of them. People creating new config headers could easily trip over this.
.

Reply:
@ceolin the main config files are generally ones you don't want to mess with.  You can fine tune those by adding your own application specific overrides.

Reply:
well, some of these configs contains changes to allows fine tuning on build. Some of them like `config-tls-generic.h` does not even exist in mbedTLS source. What I'm missing ?

Anyway, I don't want bring noise to this PR because this implementation is better IMHO :)

Reply:
@ceolin that was just an off the cuff statement.  This reminded me of the mqtt demo sample we added where I had specified specific config overrides in the application source directory.